src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +111,66 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _apiVersion);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _apiVersion);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _apiVersion);
                     return;
             }
         }
 
+        internal static string AppendOrReplaceApiVersion(string uri, string? apiVersion)
+        {
+            if (apiVersion != null)
+            {
+                var uriSpan = uri.AsSpan();
+                var apiVersionParamSpan = ApiVersionParam.AsSpan();
+                var apiVersionIndex = uriSpan.IndexOf(apiVersionParamSpan);
+                if (apiVersionIndex == -1)
+                {
+                    var concatSymbol = uriSpan.IndexOf('?') > -1 ? "&" : "?";
+                    return $"{uri}{concatSymbol}api-version={apiVersion}";
```

Comment:
Since the goal of this method seems to be partly performance, I would double check that the compiler (at least Release builds) effectively turns this into a `string.Concat` call. Seems it should, but if using `string.Format` or related APIs, those can be very expensive (lots of type checks and other conditions on operands).

Reply:
I think you're right and a benchmark test showed similar result for the two:
| Method |     Mean |     Error |    StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|------- |---------:|----------:|----------:|-------:|------:|------:|----------:|
| Format | 30.19 ns | 0.5622 ns | 0.4984 ns | 0.0135 |     - |     - |     112 B |
| Concat | 30.34 ns | 0.3059 ns | 0.2862 ns | 0.0135 |     - |     - |     112 B |